GradeCenter.Tests/AttendanceServiceTests.cs

```diff
@@ -0,0 +1,190 @@
+
+
+            schoolMock.As<IQueryable<School>>().Setup(m => m.Provider).Returns(schools.Provider);
+            schoolMock.As<IQueryable<School>>().Setup(m => m.Expression).Returns(schools.Expression);
+            schoolMock.As<IQueryable<School>>().Setup(m => m.ElementType).Returns(schools.ElementType);
+            schoolMock.As<IQueryable<School>>().Setup(m => m.GetEnumerator()).Returns(schools.GetEnumerator());
+
+
+            _dbMock.Setup(x => x.Schools).Returns(schoolMock.Object);
+
+            // Set up the database mock to return the DbSet mock for Discipline
+            var disciplineMock = new Mock<DbSet<Discipline>>();
+
+            var discipline = new Discipline
+            {
+                Id = Guid.Parse("aae2d3d8-602f-46e1-b982-3097361f5207"),
+                IsActive = true,
+                OccuranceDay = DayOfWeek.Monday,
+                Name = "Math",
+                OccuranceTime = TimeSpan.MinValue
+            };
+
+            var disciplines = new List<Discipline>() { discipline }.AsQueryable();
+
+            disciplineMock.As<IQueryable<Discipline>>().Setup(m => m.Provider).Returns(disciplines.Provider);
+            disciplineMock.As<IQueryable<Discipline>>().Setup(m => m.Expression).Returns(disciplines.Expression);
+            disciplineMock.As<IQueryable<Discipline>>().Setup(m => m.ElementType).Returns(disciplines.ElementType);
+            disciplineMock.As<IQueryable<Discipline>>().Setup(m => m.GetEnumerator()).Returns(disciplines.GetEnumerator());
+
+            _dbMock.Setup(x => x.Disciplines).Returns(disciplineMock.Object);
+
+            // Set up the database mock to return the DbSet mock for Attendance
+            var attendanceMock = new Mock<DbSet<Attendance>>();
+
+            var Attendance = new Attendance
+            {
+                Id = Guid.Parse("3144e6f5-9f45-452a-8914-8e7decbdf271"),
+                Date = DateTime.Now,
+                IsActive = true,
+                HasAttended = false,
+                Discipline = _dbMock.Object.Disciplines.ToList().FirstOrDefault(),
+                Student = _dbMock.Object.AspNetUsers.ToList().FirstOrDefault()
+            };
+
+            var attendances = new List<Attendance>() { Attendance }.AsQueryable();
+
+            attendanceMock.As<IQueryable<Attendance>>().Setup(m => m.Provider).Returns(attendances.Provider);
+            attendanceMock.As<IQueryable<Attendance>>().Setup(m => m.Expression).Returns(attendances.Expression);
+            attendanceMock.As<IQueryable<Attendance>>().Setup(m => m.ElementType).Returns(attendances.ElementType);
+            attendanceMock.As<IQueryable<Attendance>>().Setup(m => m.GetEnumerator()).Returns(attendances.GetEnumerator());
+
+            _dbMock.Setup(x => x.Attendances).Returns(attendanceMock.Object);
+
+            _attendanceService = new AttendanceService(_dbMock.Object);
+        }
+
+        [Fact]
+        public async Task Read_Attendances_ReturnsEntries()
```

Comment:
This should be named `GetAllAttendances_ReturnsCorrectData()` instead, as the current name doesn't suggest what concretely we are testing. Have in mind that we usually have multiple get methods in 1 controller and this one wont be an exception. (They'll be added in my story)